### PR TITLE
Chore: Enable gocyclo linter

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Enable [gocyclo](https://github.com/fzipp/gocyclo) linter, which detects overly complex Go functions.

In my opinion, it's very useful since it prevents too long functions, which are really difficult to follow and understand (the old rule of thumb from Code Complete is that a function should generally not exceed the height of the screen more or less). Some functions become long for a reason (usually switch statements), and I've marked them with a `nolint` directive.

**NB:** The changes, although just refactoring, are a bit of a mouthful taken altogether, I know. I can break up into smaller PRs if it would help reviewers!

**Update:** Something breaks on login, so I will split this into several PRs and check more thoroughly to see what goes wrong (probably something in the settings parsing).

### CODE REVIEW

File: pkg/tsdb/sqleng/sql_engine.go [hunk 1]
```diff
@@ -276,45 +276,61 @@ func (e *sqlQueryEndpoint) transformToTable(query *tsdb.Query, rows *core.Rows,
 	return nil
 }
 
-func (e *sqlQueryEndpoint) transformToTimeSeries(query *tsdb.Query, rows *core.Rows, result *tsdb.QueryResult, tsdbQuery *tsdb.TsdbQuery) error {
-	pointsBySeries := make(map[string]*tsdb.TimeSeries)
-	seriesByQueryOrder := list.New()
-
+func newProcessCfg(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery, rows *core.Rows) (*processCfg, error) {
```
Comment (marefr) [inline pkg/tsdb/sqleng/sql_engine.go:279]: Not sure I agree this gets less complex. Maybe easier for now to add `//nolint: gocyclo` to transformToTimeSeries?

**Review by marefr — COMMENTED** [review]
Looks good. Added a comment. Guess this is related to the other settings PR why I only leaving a comment for now?

**Comment by aknuds1** [comment]
@marefr I'll be reworking this PR, after having merged the settings refactoring PR. It's been split into smaller PRs and can now be brought up-to-date again.

**Review by marefr — CHANGES_REQUESTED** [review]
Seems to be a couple places left to fix.

**Review by marefr — APPROVED** [review]
LGTM, but need to resolve conflicts

**Pull request merged**